packages/grafana-runtime/src/utils/DataSourceBase.ts

```diff
@@ -0,0 +1,59 @@
+import { Observable } from 'rxjs';
+
+import {
+  DataSourceApi,
+  DataSourceInstanceSettings,
+  ScopedVars,
+  AdHocVariableFilter,
+  TestDataSourceResponse,
+  DataQueryRequest,
+  DataQueryResponse,
+} from '@grafana/data';
+import { DataQuery, DataSourceJsonData } from '@grafana/schema';
+
+/**
+ * This class provides generic default implementations for data source plugin methods.
+ * Extend this class to implement a data source plugin.
+ *
+ * @public
+ */
+export abstract class DataSourceBase<
```

Comment:
Maybe move this to `grafana/data` to be placed togheter with the other ds types.